src/python/pants/backend/jvm/tasks/jvm_platform_analysis.py

```diff
@@ -181,8 +185,45 @@ def validate_platform_dependencies(self):
     """
     conflicts = []
 
+    jvm_platform_map = {}
+
+    class JvmPlatformAssignment(datatype([
+        ('from_upstream', bool),
+        ('source', SubclassesOf(JvmTarget)),
+        'version',
+    ])):
+
+      @property
+      def transitive_assignment_description(self):
+        if self.from_upstream:
```

Comment:
Hm... so, we went the opposite direction in #6284: namely that when a target does not specify constraints, we always use the repo-wide default.

For consistency, it might be good to do that here as well.

Reply:
Upon reading your comment I realized this diff is actually completely wrong, as it would have allowed explicitly-specified targets to depend on unspecified targets (which in this diff are lazily converted to the default jvm platform) -- so the behavior described in the OP is correct and expected.